src/drawing.rs

```diff
@@ -80,9 +88,51 @@ impl Drawing {
     pub fn attr(&self, key: &str, def: &str) -> String {
         self.attrs.get(key).unwrap_or(&def.to_string()).clone()
     }
+
+    pub fn refdes(&self) -> &Option<TextBox> {
+        &self.refdes
+    }
+
+    pub fn value(&self) -> &Option<TextBox> {
+        &self.value
+    }
 }
 
 // Private methods
 impl Drawing {
+    fn create_textbox(&self, prefix: &str, elements: &mut SvgHash) -> Option<TextBox> {
+        let id = elements.keys().find(|&id| id.starts_with(prefix));
+        let id = match id {
```

Comment:
Isn't better to use `if let` statement here?

Reply:
I tried to reduce nesting this way, but probably it was better to use `if let` here.
Anyway, after reworking all that `ref_des`/`value`/`elements` stuff these lines are not needed any more. :slightly_smiling_face: